### DISCUSSION TITLE
Variable update frequency

### DISCUSSION THREAD

**Discussion by rj-040** [vasqued2/ha-teamtracker#54] [Ideas]
Hello,

Firstly, I have to thank the developers. Ha-teamtracker is great.

I'm not sure, but as I understand the interface, ha-teamtracker polls the servers every 120 seconds by default.
I base this off the field "Update Timeout (in seconds) 120" when setting up the integration.

My take:
Between games, 120 seconds is a bit too often. But during games, 120 seconds can be a bit too seldom.

Suggestion: 
Have variable updates. Let's say, between games, poll the server once every 2 hours.
But during games update every 30 seconds.
Could even be that when we're getting close to game-start, the update interval should be something in-between. Perhaps 2 hours before game, poll every 2 minutes, to see if there are any changes.

Caveat: This setting might be a timeout, not an update interval. If so, this topic is a bit wrong. But perhaps relevant anyway? Not sure how updates are done.

**Comment by vasqued2** [comment]
CONF_TIMEOUT/DEFAULT_TIMEOUT is legacy from the original ha-nfl sensor.  Since it's part of the schema, pulling it out would break backward compatibility.

In reality, the sensor behaves as you suggested.  It updates at 10 minute intervals until 20 minutes before game time, then updates every 5 seconds until the the game is over.

**Comment by rj-040** [comment]
Ah, I see. Great stuff. Thank you for explaining. 
Perhaps hide it from the GUI, and just set the legacy setting in the background?
And perhaps a short text explaining that it is legacy, and now update as you stated here.
But not important, just a suggestion.

Keep up the great work.

**Comment by vasqued2** [comment]
Oh, you mean when you set up the sensor.  I completely forgot about that.  I’ll see what I can do there.  Good call.
